agent-core/src/main/java/com/newrelic/agent/android/HttpHeaders.java

```diff
@@ -0,0 +1,47 @@
+package com.newrelic.agent.android;
+
+import static com.newrelic.agent.android.util.Constants.ApolloGraphQLHeader.OPERATION_ID;
+import static com.newrelic.agent.android.util.Constants.ApolloGraphQLHeader.OPERATION_NAME;
+import static com.newrelic.agent.android.util.Constants.ApolloGraphQLHeader.OPERATION_TYPE;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class HttpHeaders {
+
+    private static HttpHeaders instance = null;
```

Comment:
If you're using a singleton, follow the pattern used elsewhere in the agent and use an [AtomIcReference<T>](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html).